src/toxcore/dht/server/mod.rs

```diff
@@ -640,31 +699,33 @@ impl Server {
             Packet::OnionRequest2(packet) => Box::new(self.handle_onion_request_2(&packet, addr)),
             Packet::OnionAnnounceRequest(packet) => Box::new(self.handle_onion_announce_request(packet, addr)),
             Packet::OnionDataRequest(packet) => Box::new(self.handle_onion_data_request(packet)),
-            Packet::OnionResponse3(packet) => Box::new(self.handle_onion_response_3(packet)),
-            Packet::OnionResponse2(packet) => Box::new(self.handle_onion_response_2(packet)),
-            Packet::OnionResponse1(packet) => Box::new(self.handle_onion_response_1(packet)),
-            Packet::BootstrapInfo(packet) => Box::new(self.handle_bootstrap_info(&packet, addr)),
+            Packet::OnionResponse3(packet) => Box::new(self.handle_onion_response_3(packet)
+                .map_err(|e| HandlePacketError::from(e))),
```

Comment:
Perhaps move these `map`s to `handle_onion_response_*` to be consistent with other handling packets methods?